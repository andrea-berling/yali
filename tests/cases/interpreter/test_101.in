// This program creates a function that returns another function
// and uses it to filter a list of numbers
fun makeFilter(min) {
  fun filter(n) {
    if (n < min) {
      return false;
    }
    return true;
  }
  return filter;
}
// This function applies a function to a list of numbers
fun applyToNumbers(f, count) {
  var n = 0;
  while (n < count) {
    if (f(n)) {
      print n;
    }
    n = n + 1;
  }
}
var greaterThanX = makeFilter(30);
var greaterThanY = makeFilter(74);
print "Numbers >= 30:";
applyToNumbers(greaterThanX, 30 + 3);

print "Numbers >= 74:";
applyToNumbers(greaterThanY, 74 + 3);
