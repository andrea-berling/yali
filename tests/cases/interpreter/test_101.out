Numbers >= 30:
30
31
32
Numbers >= 74:
74
75
76
